test/e2e/pipeline/utils.sh

```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 #shellcheck disable=SC1091
 
+set -e           # Fail in case of error
+set -o nounset   # Fail if undefined variables are used
+set -o pipefail  # Fail if one of the piped commands fails
```

Comment:
Are you sure it is necessary here? I would suggest to add error handling only in files that import utilities.